src/components/OpenWith/OpenWith.js

```diff
@@ -208,22 +217,85 @@ class OpenWith extends PureComponent<Props, State> {
     }
 
     /**
-     * Renders the Open With element
+     * Click handler when an integration is clicked
+     *
+     * @private
+     * @return {void}
+     */
+    onIntegrationClick(): void {
+        /* no-op */
+    }
+
+    /**
+     * Gets the tooltip text for the OpenWith button
+     *
+     * @private
+     * @return {?string | ?Element}
+     */
+    getTooltip(): ?string | ?Element {
+        const { fetchError, isLoading, integrations }: State = this.state;
+        const displayIntegration = this.getDisplayIntegration();
+        if (isLoading) {
+            return null;
+        } else if (fetchError) {
+            return <FormattedMessage {...messages.errorOpenWithDescription} />;
+        } else if (displayIntegration) {
+            return displayIntegration.displayDescription;
+        } else if (Array.isArray(integrations) && integrations.length > 1) {
+            return (
+                <FormattedMessage {...messages.defaultOpenWithDescription} />
+            );
+        }
+
+        return <FormattedMessage {...messages.emptyOpenWithDescription} />;
+    }
+
+    /**
+     * Gets a display integration, if available, for the Open With button
+     *
+     * @private
+     * @return {?Integration}
+     */
+    getDisplayIntegration(): ?Integration {
+        const { integrations }: State = this.state;
+        // We only consider an integration a display integration if is the only integration in our state
+        return Array.isArray(integrations) && integrations.length === 1
+            ? integrations[0]
+            : null;
+    }
+
+    /**
+     * Render the Open With element
      *
      * @private
      * @inheritdoc
      * @return {Element}
      */
     render() {
         const { language, messages: intlMessages }: Props = this.props;
-        const { integrations }: State = this.state;
-        // Placeholder
+        const { fetchError: error, isLoading, extension }: State = this.state;
+
+        const displayIntegration = this.getDisplayIntegration();
+
         return (
-            integrations && (
-                <Internationalize language={language} messages={intlMessages}>
-                    <button> Open With </button>
-                </Internationalize>
-            )
+            <Internationalize language={language} messages={intlMessages}>
+                <div id={this.id} className="be bcow">
```

Comment:
Should the class be `be-bcow`?

Reply:
The other elements have been adding them separately